Mathlib/Order/LiminfLimsup.lean

```diff
@@ -0,0 +1,1242 @@
+  simp only [not_lt, le_zero_iff, infᵢ_infᵢ_eq_left, add_zero, infᵢ_nat_gt_zero_eq, left_eq_inf]
+  refine' (infᵢ_le (fun i => ⨆ j, (f^[j + (i + 1)]) a) 0).trans _
+  simp only [zero_add, Function.comp_apply, supᵢ_le_iff]
+  exact fun i => le_supᵢ (fun i => (f^[i]) a) (i + 1)
+#align filter.complete_lattice_hom.apply_limsup_iterate Filter.CompleteLatticeHom.apply_limsup_iterate
+
+/-- If `f : α → α` is a morphism of complete lattices, then the liminf of its iterates of any
+`a : α` is a fixed point. -/
+theorem CompleteLatticeHom.apply_liminf_iterate (f : CompleteLatticeHom α α) (a : α) :
+    f (liminf (fun n => (f^[n]) a) atTop) = liminf (fun n => (f^[n]) a) atTop :=
+  apply_limsup_iterate (CompleteLatticeHom.dual f) _
+#align filter.complete_lattice_hom.apply_liminf_iterate Filter.CompleteLatticeHom.apply_liminf_iterate
+
+variable {f g : Filter β} {p q : β → Prop} {u v : β → α}
+
+theorem blimsup_mono (h : ∀ x, p x → q x) : blimsup u f p ≤ blimsup u f q :=
+  infₛ_le_infₛ fun a ha => ha.mono <| by tauto
+#align filter.blimsup_mono Filter.blimsup_mono
+
+theorem bliminf_antitone (h : ∀ x, p x → q x) : bliminf u f q ≤ bliminf u f p :=
+  supₛ_le_supₛ fun a ha => ha.mono <| by tauto
+#align filter.bliminf_antitone Filter.bliminf_antitone
+
+theorem mono_blimsup' (h : ∀ᶠ x in f, p x → u x ≤ v x) : blimsup u f p ≤ blimsup v f p :=
+  infₛ_le_infₛ fun _ ha => (ha.and h).mono fun _ hx hx' => (hx.2 hx').trans (hx.1 hx')
+#align filter.mono_blimsup' Filter.mono_blimsup'
+
+theorem mono_blimsup (h : ∀ x, p x → u x ≤ v x) : blimsup u f p ≤ blimsup v f p :=
+  mono_blimsup' <| eventually_of_forall h
+#align filter.mono_blimsup Filter.mono_blimsup
+
+theorem mono_bliminf' (h : ∀ᶠ x in f, p x → u x ≤ v x) : bliminf u f p ≤ bliminf v f p :=
+  supₛ_le_supₛ fun _ ha => (ha.and h).mono fun _ hx hx' => (hx.1 hx').trans (hx.2 hx')
+#align filter.mono_bliminf' Filter.mono_bliminf'
+
+theorem mono_bliminf (h : ∀ x, p x → u x ≤ v x) : bliminf u f p ≤ bliminf v f p :=
+  mono_bliminf' <| eventually_of_forall h
+#align filter.mono_bliminf Filter.mono_bliminf
+
+theorem bliminf_antitone_filter (h : f ≤ g) : bliminf u g p ≤ bliminf u f p :=
+  supₛ_le_supₛ fun _ ha => ha.filter_mono h
+#align filter.bliminf_antitone_filter Filter.bliminf_antitone_filter
+
+theorem blimsup_monotone_filter (h : f ≤ g) : blimsup u f p ≤ blimsup u g p :=
+  infₛ_le_infₛ fun _ ha => ha.filter_mono h
+#align filter.blimsup_monotone_filter Filter.blimsup_monotone_filter
+
+
+-- @[simp] -- Porting note: simp_nf linter, lhs simplifies, added _aux version below
+theorem blimsup_and_le_inf : (blimsup u f fun x => p x ∧ q x) ≤ blimsup u f p ⊓ blimsup u f q :=
+  le_inf (blimsup_mono <| by tauto) (blimsup_mono <| by tauto)
+#align filter.blimsup_and_le_inf Filter.blimsup_and_le_inf
+
+@[simp]
+theorem bliminf_sup_le_inf_aux :
+  (blimsup u f fun x => p x ∧ q x) ≤ blimsup u f p
+    ∧ (blimsup u f fun x => p x ∧ q x) ≤ blimsup u f q := by
+  rw [←le_inf_iff]
+  exact blimsup_and_le_inf
```

Comment:
```suggestion
-- Porting note: new lemma
@[simp]
theorem bliminf_sup_le_inf_aux :
  (blimsup u f fun x => p x ∧ q x) ≤ blimsup u f p
    ∧ (blimsup u f fun x => p x ∧ q x) ≤ blimsup u f q := by
  rw [←le_inf_iff]
  exact blimsup_and_le_inf
```
I think you can make simp even more powerful by splitting this into two lemmas: one for the left hand side and one for the right hand side.